src/compiler/binder.ts

```diff
@@ -1276,6 +1284,22 @@ namespace ts {
             activeLabels!.pop();
         }
 
+        function isDottedName(node: Expression): boolean {
+            return node.kind === SyntaxKind.Identifier || node.kind === SyntaxKind.PropertyAccessExpression && isDottedName((<PropertyAccessExpression>node).expression);

```

Comment:
Is there a reason not to include `this` and `super` in property access expressions?

Reply:
I *think* that would be okay, but I'll have to convince myself it can't trigger circularities in control flow analysis.